modin/numpy/arr.py

```diff
@@ -704,50 +741,99 @@ def mean(self, axis=None, dtype=None, out=None, keepdims=None, *, where=True):
             if dtype is not None
             else (out.dtype if out is not None else self.dtype)
         )
+        out_type = out_dtype if not hasattr(out_dtype, "type") else out_dtype.type
```

Comment:
```suggestion
        out_type = getattr(out_dtype, "type", out_dtype)
```
😉 
does the same, works slightly faster (because `hasattr()` internally does the `getattr()` thing and returns True if no error happened)

Reply:
Fixed!